src/Containers/OhmmsPETE/OhmmsMatrix.h

```diff
@@ -163,14 +164,36 @@ class Matrix
   }
 
   // Assignment Operators
+  /// From another Matrix with matching type.
   inline This_t& operator=(const This_t& rhs)
   {
     resize(rhs.D1, rhs.D2);
+    // I don't understand why it would be desirable to have this compile
+    // but just do the resize and not the assigment, just seems like a surprising foot gun.
```

Comment:
A good catch. Users can abuse it for making matching matrices without the need of copying over values. Such use cases should not call operator `=` .
Needs a revisit when we scrutinize the operator `=` overloading.